opcua_tools/value_parser.py

```diff
@@ -247,6 +248,26 @@ def parse_singular_value(val, tagtype):
     raise NotImplementedError(tagtype)
 
 
+@functools.cache
+def cached_parse_nodeid(
+    nodeidstr: str,
+) -> Tuple(int, NodeIdType, str):

```

Comment:
I see that tests are failing because of `Tuple`. You can import it from the `typing` library (the old way) or simply use `tuple` instead (the Python 3.9+ way). When it comes to me, I'd go with the second option.